app/models/project.server.ts

```diff
@@ -197,121 +210,143 @@ export async function getProjects(where: ProjectWhereInput) {
-        data: {
-          practicedSkills: { set: [] },
-          role: { set: [] },
-        },
-        include: {
-          practicedSkills: true,
-          role: true,
-        },
-      })
+      await db.$queryRaw`DELETE FROM "_ProjectMembersToSkills" WHERE "A" = ${previousMember.id}`;
+      await db.$queryRaw`DELETE FROM "_DisciplinesToProjectMembers" WHERE "B" = ${previousMember.id}`;
       // Makes all the actual updates to the projectMember
       await db.projectMembers.update({
-        where: { id: data.projectMembers[index].id },
+        where: { id: previousMember.id },
         data: {
-          hoursPerWeek: data.projectMembers[index].hoursPerWeek,
-          role: { connect: rolesArrayConnect },
-          active: data.projectMembers[index].active,
-          practicedSkills: { connect: practicedSkillsArrayConnect },
+          hoursPerWeek: projectMembers[i].hoursPerWeek,
+          role: { connect: projectMembers[i].role },
+          active: projectMembers[i].active,
+          practicedSkills: { connect: projectMembers[i].practicedSkills },
         },
         include: {
           practicedSkills: true,
         },
-      })
+      });
     } else {
       await db.projectMembers.create({
         data: {
-          project: { connect: { id } },
-          profile: { connect: { id: data.projectMembers[index].profileId } },
-          hoursPerWeek: data.projectMembers[index].hoursPerWeek,
-          role: { connect: rolesArrayConnect },
-          practicedSkills: { connect: practicedSkillsArrayConnect },
+          project: { connect: { id: projectId } },
+          profile: { connect: { id: projectMembers[i].profileId } },
+          hoursPerWeek: projectMembers[i].hoursPerWeek,
+          role: { connect: projectMembers[i].role },
+          practicedSkills: { connect: projectMembers[i].practicedSkills },
         },
-      })
+      });
     }
   }
 
-  // Delete members who are no longer active
-  for (let j = 0; j < data.existedMembers?.length; j++) {
-    if (!activeMembers.includes(data.existedMembers[j])) {
-      await db.projectMembers.deleteMany({ where: { id: data.existedMembers[j] } })
+  // Delete previous members who are no longer in activeMembers array of ids
+  for (let j = 0; j < previousMembers.length; j++) {
+    if (!activeMembers.includes(previousMembers[j].profileId)) {
+      await db.projectMembers.deleteMany({
+        where: { profileId: previousMembers[j].profileId, projectId },
+      });
     }
   }
+}
+
+export async function updateProjects(
+  id: string,
+  isAdmin: boolean,
+  data: {
+    repoUrls?: any;
+    projectMembers: any;
+    relatedProjects?: any;
+    projectStatus?: any;
+    owner: any;
+    skills?: any;
+    disciplines?: any;
+    labels?: any;
+    innovationTiers?: any;
+  }
+) {
+  //validate if the user have permissions (team member or owner of the project)
+  if (!isAdmin) validateIsTeamMember(id, data.projectMembers, data.owner.id);
```

Comment:
Same here for the validation, and i think that the fact that we have this code repeated is a good reason to group the Business rules together.